--- lib/functions.sh
+++ lib/functions.sh
@@ -1,6 +1,6 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
-. ${MPATH}/lib/colors.sh
+source ${MPATH}/colors.sh
 
 _green_msg(){
     echo "${GREEN}$@${NC}"
diff --git i/m w/m
index fe903d6..1d486a8 100755
--- m
+++ m
@@ -1,25 +1,16 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
-pushd `dirname $0` > /dev/null
-export MPATH=`pwd -P`
-popd > /dev/null
+export MPATH="${M_CLI_PATH:-/opt/local/libexec/m-cli}"
 
-. ${MPATH}/lib/colors.sh
-. ${MPATH}/lib/functions.sh
-
-
-usage(){
+source ${MPATH}/functions.sh
 
+usage() {
     cat <<__EOF__
 
   Swiss Army Knife for Mac OS X ! 
 
 
-usage:  m [OPTIONS] COMMAND [help]
-
-    OPTIONS
-        --update        update m-cli to the latest version
-        --uninstall     uninstall m-cli
+usage:  m COMMAND [help]
 
     COMMANDS:
         help
@@ -31,27 +22,9 @@ __EOF__
     exit 1
 }
 
-case $1 in
-    --update)
-        ${MPATH}/install.sh
-        exit 0
-        ;;
-    --uninstall)
-        confirm "Do you want to uninstall m-cli? [y/n]: " \
-            && sudo rm -rf ${MPATH} 2>/dev/null \
-            && sudo rm -f "/etc/paths.d/m-cli" 2>/dev/null \
-            && echo "Done !"
-        exit 0
-        ;;
-esac
-
-
 COMMAND=$1; shift;
 
 [ ! -f ${MPATH}/plugins/${COMMAND} ] && usage
 ${MPATH}/plugins/${COMMAND} $@
 
-
-
 # vim: set ts=4 sw=4 softtabstop=4 expandtab
-
